Sequencer.cs

```diff
@@ -16,39 +16,99 @@ namespace WordSequence
-            FileStream configStream = File.OpenRead(GetConfigurationPath());
-            try
+            /* pass "false" to GetConfigurationPath to default to the global
+             * config when user config not found
+             */
+            string configFile = GetConfigurationPath(false);
+            if (null != configFile && File.Exists(configFile))
             {
-                return (PasswordSequenceConfiguration)serializer.Deserialize(XmlReader.Create(configStream));
+                /* TODO: replace xsd path with local path instead of web path
+                 * that could change?
+                 */
+                XmlSerializer serializer = new XmlSerializer(typeof(PasswordSequenceConfiguration), "http://quasivirtuel.com/PasswordSequenceConfiguration.xsd");
+                FileStream configStream = File.OpenRead(configFile);
+                try
+                {
+                    return (PasswordSequenceConfiguration)serializer.Deserialize(XmlReader.Create(configStream));
+                }
+                finally
+                {
+                    configStream.Close();
+                }
             }
-            finally
+            else
             {
-                configStream.Close();
+                /* Config file not found; create empty config */
+                return new PasswordSequenceConfiguration(true);
+                /* TODO: pop up an error message or something? */
             }
         }
 
         public void Save(PasswordSequenceConfiguration configuration)
         {
-            XmlSerializer serializer = new XmlSerializer(typeof(PasswordSequenceConfiguration), "http://quasivirtuel.com/PasswordSequenceConfiguration.xsd");
-            FileStream configStream = File.Open(GetConfigurationPath(), FileMode.Create);
-            try
+            /* pass "true" to GetConfigurationPath to default to the user config
+             * even when it doesn't exist yet; we'll create it here
+             */
+            string configFile = GetConfigurationPath(true);
+            if (null != configFile)
             {
-                serializer.Serialize(configStream, configuration);
-            }
-            finally
-            {
-                configStream.Close();
+                XmlSerializer serializer = new XmlSerializer(typeof(PasswordSequenceConfiguration), "http://quasivirtuel.com/PasswordSequenceConfiguration.xsd");
+
+                /* create the config file's directory if needed */
+                Directory.CreateDirectory(Path.GetDirectoryName(configFile));
+
+                /* open the file for writing, creating a new one if needed */
+                FileStream configStream = File.Open(configFile, FileMode.Create);
+
+                try
+                {
+                    serializer.Serialize(configStream, configuration);
+                }

```

Comment:
Nope, because this would be in an "else" block, if it is needed at all. I'm conflicted about whether it is needed at all. Should I remove the comment? Or would you prefer a popup saying something like "could not locate configuration file location"?